### DISCUSSION TITLE
Can i restrict a backend file upload based on size or extension?

### DISCUSSION THREAD

**Discussion by mpiorowski** [supabase/supabase#1372] [Questions] [answered]
Is there a way to set a rules for file upload using supabase? So the user cannot upload anything he want?

**Comment by inian** [comment, marked as answer]
Hi @mpiorowski, this is not possible yet. We are tracking the issue [here](https://github.com/supabase/storage-api/issues/16). Current default is 50 MB.

**Reply by mpiorowski** [reply]
Thank you for response :)
